pkg/proxy/handlers/spacelister.go

```diff
@@ -160,17 +163,55 @@ func (s *SpaceLister) ListUserWorkspaces(ctx echo.Context) ([]toolchainv1alpha1.
 		// account exists but the compliant username is not set yet, meaning it has not been fully provisioned yet, so return an empty list
 		return []toolchainv1alpha1.Workspace{}, nil
 	}
-
 	murName := signup.CompliantUsername
+
+	if workspaceName != "" {
+		// if a workspace was provided
+		// recursively get all the spacebindings, starting from this workspace and going up to the parent workspaces till the "root" of the workspace tree.
+		return s.hierarchicalWorkspaceSearch(ctx, workspaceName, signup)
+	}
+
+	// get all spacebindings with given mur since no workspace was provided
 	spaceBindings, err := s.listSpaceBindingsForUser(murName)
```

Comment:
it this case it should go "down" in the structure of sub-spaces and collect all sub-spaces the user's SpaceBinding is inherited for.
It can be done as a separate story ;-)

Reply:
🤔 
ok, this seems a bit different use case. I guess we could:
- get all spacebindings
- for each space from the spacebindings get the list of subSpaces (using the `toolchainv1alpha1.ParentSpaceLabelKey` label)

So yeah I think this is a separate task 😅 . Let me create a story for it.